hnn_core/cell.py

```diff
@@ -220,6 +224,55 @@ def __init__(self, L, diam, Ra, cm, end_pts=None):
     def __repr__(self):
         return f'L={self.L}, diam={self.diam}, cm={self.cm}, Ra={self.Ra}'
 
+    def __eq__(self, other):
+        if not isinstance(other, Section):
+            return NotImplemented
```

Comment:
Oh I think this is why the pytest.raises test didn't work. You're returning `NotImplemented`, but you actually need to do `raise NotImplemented` (check out how the other errors get raised)